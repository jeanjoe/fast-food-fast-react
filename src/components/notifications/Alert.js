import React from "react";
import { Message } from "semantic-ui-react";

const Alert = ({ message, type }) => {
  return (
    <div>
      {type === "positive" ? (
        <Message positive>
          <Message.Header as="h5"> {message} </Message.Header>
        </Message>
      ) : (
        <Message negative>
          <Message.Header as="h5"> {message} </Message.Header>
        </Message>
      )}
    </div>
  );
};

export default Alert;
