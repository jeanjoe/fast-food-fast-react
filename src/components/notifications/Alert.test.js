import React from "react";
import expect from "expect";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Alert from "./Alert";
configure({ adapter: new Adapter() });

describe("Test article component", () => {
  const props = {
    message: 'error',
    type: 'positive',
  };
  const wrapper = shallow(<Alert {...props} />);

  it("Matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
