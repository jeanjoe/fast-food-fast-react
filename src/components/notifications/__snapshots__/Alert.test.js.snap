// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test article component Matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    message="error"
    type="positive"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Message
        positive={true}
      >
        <MessageHeader
          as="h5"
        >
           
          error
           
        </MessageHeader>
      </Message>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <MessageHeader
          as="h5"
        >
           
          error
           
        </MessageHeader>,
        "positive": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h5",
          "children": Array [
            " ",
            "error",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "error",
          " ",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Message
          positive={true}
        >
          <MessageHeader
            as="h5"
          >
             
            error
             
          </MessageHeader>
        </Message>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MessageHeader
            as="h5"
          >
             
            error
             
          </MessageHeader>,
          "positive": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h5",
            "children": Array [
              " ",
              "error",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "error",
            " ",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
